File: TodayFitComplete_Backend/src/main/java/com/SJY/TodayFitComplete_Backend/dto/board/BoardWriteResponse.java
package com.SJY.TodayFitComplete_Backend.dto.board;

import com.SJY.TodayFitComplete_Backend.entity.board.Board;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * 게시글 작성 응답 DTO.
 */
@Getter
@Setter
@NoArgsConstructor
public class BoardWriteResponse {

    private Long boardId;
    private String title;
    private String content;
    private String writerName;
    private String createdDate;

    @Builder
    public BoardWriteResponse(Long boardId, String title, String content, String writerName, String createdDate) {
        this.boardId = boardId;
        this.title = title;
        this.content = content;
        this.writerName = writerName;
        this.createdDate = createdDate;
    }

    public static BoardWriteResponse fromEntity(Board board) {
        return BoardWriteResponse.builder()
                .boardId(board.getId())
                .title(board.getTitle())
                .content(board.getContent())
                .writerName(board.getMember().getNickname())
                .createdDate(board.getCreatedDate())
                .build();
    }
}

File: TodayFitComplete_Frontend/app/src/main/java/kr/ac/kumoh/ce/s20190633/todayfitcomplete_frontend/Dto/Board/BoardWriteResponse.kt
package kr.ac.kumoh.ce.s20190633.todayfitcomplete_frontend.Dto.Board

// BoardWriteResponse 클래스: 게시글 작성 후 서버에서 반환되는 응답 데이터를 나타냅니다.
data class BoardWriteResponse(
    val boardId: Long,                   // 생성된 게시글의 고유 ID
    val title: String,                   // 게시글 제목
    val content: String,                 // 게시글 내용
    val writerName: String,              // 작성자 이름
    val createdDate: String              // 작성일
)